apps/server/src/modules/school/service/school.service.spec.ts

```diff
@@ -219,74 +232,106 @@ describe('SchoolService', () => {
 		});
 	});
 
-	describe('save', () => {
-		it('should call the repo', async () => {
-			const school: SchoolDO = new SchoolDO({ id: 'id', name: 'name' });
+	describe('save is called', () => {
+		describe('when school is given', () => {
+			it('should call the repo', async () => {
+				const school: SchoolDO = new SchoolDO({ id: 'id', name: 'name' });
 
-			await schoolService.save(school);
+				await schoolService.save(school);
 
-			expect(schoolRepo.save).toHaveBeenCalledWith(school);
-		});
+				expect(schoolRepo.save).toHaveBeenCalledWith(school);
+			});
 
-		it('should return a do', async () => {
-			const school: SchoolDO = new SchoolDO({ id: 'id', name: 'name' });
-			schoolRepo.save.mockResolvedValue(school);
+			it('should return a do', async () => {
+				const school: SchoolDO = new SchoolDO({ id: 'id', name: 'name' });
+				schoolRepo.save.mockResolvedValue(school);
 
-			const schoolDO: SchoolDO = await schoolService.save(school);
+				const schoolDO: SchoolDO = await schoolService.save(school);
 
-			expect(schoolDO).toBeInstanceOf(SchoolDO);
+				expect(schoolDO).toBeInstanceOf(SchoolDO);
+			});
 		});
 	});
 
-	describe('setMigration', () => {
+	describe('setMigration is called', () => {
 		let testId: string;
 		let testDO: SchoolDO;
-		beforeAll(() => {
+		beforeEach(() => {
 			testId = 'migration';
 			testDO = new SchoolDO({
 				id: testId,
 				name: 'testDO',
-				oauthMigrationPossible: true,
-				oauthMigrationMandatory: true,
+				oauthMigrationPossible: new Date(),
+				oauthMigrationMandatory: new Date(),
+				oauthMigrationFinished: new Date(),
+				officialSchoolNumber: '1337',
 			});
 			schoolRepo.findById.mockResolvedValue(testDO);
 			schoolRepo.save.mockResolvedValue(testDO);
 		});
-		it('it should set the migrationflags', async () => {
-			const resp: MigrationResponse = await schoolService.setMigration(testId, true, true);
-			expect(resp.oauthMigrationPossible).toBeTruthy();
-			expect(resp.oauthMigrationMandatory).toBeTruthy();
-			expect(schoolRepo.findById).toHaveBeenCalledWith(testId);
-			expect(schoolRepo.save).toHaveBeenCalledWith(testDO);
+		describe('when migrationflags and schoolId are given', () => {
+			it('should set the migrationflags', async () => {
+				const resp: MigrationResponse = await schoolService.setMigration(testId, true, true, true);
+				expect(resp.oauthMigrationPossible).toEqual(testDO.oauthMigrationPossible);
+				expect(resp.oauthMigrationMandatory).toEqual(testDO.oauthMigrationMandatory);
+				expect(resp.oauthMigrationFinished).toEqual(testDO.oauthMigrationFinished);
+				expect(resp.enableMigrationStart).toBeTruthy();
+				expect(schoolRepo.findById).toHaveBeenCalledWith(testId);
```

Comment:
please split this into multiple tests. one checking the result, one for fetching the correct data, one for persisting.